language/diem-tools/writeset-transaction-generator/src/main.rs

```diff
@@ -29,22 +39,70 @@ enum Command {
     /// Build a custom file in templates into admin script
     #[structopt(name = "build-custom-script")]
     BuildCustomScript { script_name: String, args: String },
+    /// Cherry pick a list of modules from the stdlib to form a WriteSet.
+    #[structopt(name = "cherry-pick-modules")]
+    CherryPickModules { names: Vec<String> },
 }
 
-fn save_transaction(txn: Transaction, path: PathBuf) -> Result<()> {
-    let bytes = bcs::to_bytes(&txn).map_err(|_| format_err!("Transaction Serialize Error"))?;
+fn save_bytes(bytes: Vec<u8>, path: PathBuf) -> Result<()> {
     std::fs::write(path.as_path(), bytes.as_slice())
         .map_err(|_| format_err!("Unable to write to path"))
 }
 
 fn main() -> Result<()> {
     let opt = Opt::from_args();
-    let transaction = match opt.cmd {
-        Command::RemoveValidators { addresses } => encode_remove_validators_transaction(addresses),
-        Command::HaltNetwork => encode_halt_network_transaction(),
+    let payload = match opt.cmd {
+        Command::RemoveValidators { addresses } => encode_remove_validators_payload(addresses),
+        Command::HaltNetwork => encode_halt_network_payload(),
         Command::BuildCustomScript { script_name, args } => {
             encode_custom_script(&script_name, &serde_json::from_str(args.as_str())?)
         }
+        Command::CherryPickModules { names } => {
+            let modules = build_stdlib()
+                .into_iter()
+                .filter(|(module_name, _)| names.iter().any(|name| module_name.contains(name)))
+                .map(|(name, module)| {
+                    println!("Selecting module {:?}", name);
+                    module
+                })
+                .collect::<Vec<_>>();
+            // TODO: Add Move doctor test here to make sure the state with this new module is
```

Comment:
Just curious here, will there be a case when we are required to generate an invalid / corrupted module?

Reply:
Actually I would like to rework on this a bit more to make sure we can guard against such case!